gateway/src/task.rs

```diff
@@ -537,13 +543,29 @@ impl Task<()> for ProjectTask {
                 TaskResult::Done(_) => {
                     let _ = self.tasks.pop_front().unwrap();
                     if self.tasks.is_empty() {
+                        // Should coincide with the end of a project task started by
+                        // an API call or the ambulance.
+                        if let Some(operation) = &self.operation_name {
+                            Span::current().record("ctx.operation_name", operation);
+                            Span::current().record("shuttle.operation_name", operation);
```

Comment:
I think you can re-use the `span` field from above. It should be the same.

Reply:
It can not be borrowed explicitly inside the async block because it is moved into the 'instrument' call at the end.